### ISSUE TITLE
Updates dropped for 1 window when using `wezterm connect`

### ISSUE THREAD

**Issue by algmyr** [wez/wezterm#1841]
### What Operating System(s) are you seeing this problem on?

Linux X11

### WezTerm version

20220407-215528-d356b72c

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

On connecting to a domain with more than 1 window, the first window stops handling updates properly. It seems that if the command takes ~0.1s or longer, the output will not show until some event happens (e.g. focus event, mouse click, keyboard press).

E.g. `sleep 0.1 && echo asd` exhibits the issue, while `sleep 0.05 && echo asd` doesn't.

The problem was first observed with `ssh_domain`, but I later confirmed it purely locally with a `unix_domain`.

For `unix_domain`  it also happens on the first run of `wezterm connect unix`. Not sure if the same issue, but same symptoms.

This video shows both the symptoms on the first run, and the issue with 2+ windows.

https://user-images.githubusercontent.com/15268706/162463991-e750078e-021e-411d-8f91-39ad1135d151.mp4

### To Reproduce

1. `wezterm connect unix`
2. `super+n` to create new window
3. Close both windows
4. `wezterm connect unix`
5. Run something like `sleep 0.1 && echo asd` in the first spawned terminal, the output (and prompt) will not shop up until some event happens.

### Configuration

```
return {
  unix_domains = {
    {
      name = "unix",
    }
  },
}
```

### Expected Behavior

Command output should show immediately, not after some event happens to trigger a refresh. The behavior also shouldn't differ between the first spawned window and all other windows.

**Comment by wez** [comment]
Sounds similar to https://github.com/wez/wezterm/issues/1814 but that seemed to correlate with Wayland and not X11

**Comment by wez** [comment]
Thanks for the report: the multiple window insight was key to figuring this out, and I think it may also have resolved #1814 as well.
This is fixed now in `main` and should show up in a nightly build within ~30 minutes from now.

**Comment by algmyr** [comment]
Neat. Happy that my report was helpful. :+1:
